crates/optimism/payload/src/builder.rs

```diff
@@ -392,7 +392,7 @@ pub trait OpPayloadTransactions<Transaction>: Clone + Send + Sync + Unpin + 'sta
     ) -> impl PayloadTransactions<Transaction = Transaction>;
 }
 
-impl<T: PoolTransaction> OpPayloadTransactions<T> for () {
+impl<T: PoolTransaction + MaybeInteropTransaction> OpPayloadTransactions<T> for () {
```

Comment:
why not simply filter out any txns which aren't valid wrt interop here? not sure there is a need for a new `OpBestTransactions` type

Reply:
Problem is that tx doesn't have is_valid() method in OpBestTransactions context (or i don't see one?) 
It's accessible only in validator as i understand
We could try use client and bring OpTransactionValidator but it seems like overkill
And to check interop tx you need some kind of timestamp (to check against)

Reply:
there is the trait bound `MaybeInteropTransaction` on `T`

Reply:
Ah, got it!
The part with interop itself was working, if I check it against 0, you could check the first commit
But to check interop we also node some timestamp to check against, and that's when I started trying to include BuilderContext and things got compocated